File: python3/938.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        sum = 0
        def Dfs(node:Optional[TreeNode])->None:
            nonlocal sum
            if node is None:
               return
            if node.val < low:
                Dfs(node.right)
            if(node.val > high):
                Dfs(node.left)
            if node.val >= low and node.val <= high:
               sum += node.val
               Dfs(node.left)
               Dfs(node.right)
        Dfs(root)
        return sum

File: c/938.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int SumBST(struct TreeNode* root, int low, int high){
    if(root == NULL) return 0;
    if(root->val < low) return SumBST(root->right,low,high);
    else if(root->val > high) return SumBST(root->left,low,high);
    else return root->val + SumBST(root->left,low,high) + SumBST(root->right,low,high);
}


int rangeSumBST(struct TreeNode* root, int low, int high) {
    int sum;
    sum = SumBST(root,low,high);
    return sum;
}